backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/PostgresFieldOfLawRepositoryImpl.java

```diff
@@ -72,119 +76,57 @@ public Slice<FieldOfLaw> findAllByOrderByIdentifierAsc(Pageable pageable) {
 
   @Override
   @Transactional
-  public List<FieldOfLaw> findBySearchTerms(String[] searchTerms) {
-    if (searchTerms == null || searchTerms.length == 0) {
-      return Collections.emptyList();
+  public List<FieldOfLaw> findByCombinedCriteria(
+      String identifier, String[] descriptionSearchTerms, String[] normSearchTerms) {
+    CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+    CriteriaQuery<FieldOfLawDTO> cq = cb.createQuery(FieldOfLawDTO.class);
+    Root<FieldOfLawDTO> fieldOfLawRoot = cq.from(FieldOfLawDTO.class);
+    fieldOfLawRoot.fetch("norms", JoinType.LEFT);
+    ArrayList<Predicate> predicates = new ArrayList<>();
+
+    Predicate notationPredicate =
+        cb.equal(fieldOfLawRoot.get("notation"), cb.literal(Notation.NEW));
+    predicates.add(notationPredicate);
+
+    if (descriptionSearchTerms != null) {
+      for (String searchTerm : descriptionSearchTerms) {
+        predicates.add(
+            cb.like(cb.lower(fieldOfLawRoot.get("text")), "%" + searchTerm.toLowerCase() + "%"));
+      }
     }
-
-    List<FieldOfLawDTO> listWithFirstSearchTerm =
-        repository.findAllByNotationAndIdentifierContainingIgnoreCaseOrTextContainingIgnoreCase(
-            searchTerms[0]);
-
-    if (searchTerms.length == 1) {
-      return listWithFirstSearchTerm.stream()
-          .map(PostgresFieldOfLawRepositoryImpl::getWithNormsWithoutChildren)
-          .toList();
+    if (identifier != null) {
+      Predicate identifierPredicate =
+          cb.like(fieldOfLawRoot.get("identifier"), identifier.toUpperCase() + "%");
```

Comment:
question: For the other search params, we search by `contains` and only for this one with `startsWith`. Is that correct? (I assume the identifier is known, so it does make sense to me, just wanted to double check.)

Reply:
yup! we always assume they know at least the first two letters of the identifier they want to search in, so it made sense to me that way :)